spec/navigation_sidebar_spec.rb

```diff
@@ -0,0 +1,139 @@
+require "spec_helper"
+
+RSpec.describe GovukNavigationHelpers::RelatedNavigationSidebar do
+  def payload_for(schema, content_item)
+    generator = GovukSchemas::RandomExample.for_schema(schema, schema_type: "frontend")
+    fully_valid_content_item = generator.merge_and_validate(content_item)
+    GovukNavigationHelpers::NavigationHelper.new(fully_valid_content_item).related_navigation_sidebar
+  end
+
+  describe "#related-navigation-sidebar" do
+    it "can handle randomly generated content" do
+      generator = GovukSchemas::RandomExample.for_schema("placeholder", schema_type: "frontend")
+
+      expect { payload_for("placeholder", generator.payload) }.to_not raise_error
+    end
+
+    it "returns empty arrays if there are no related navigation sidebar links" do
+      nothing = payload_for("placeholder",
+        "details" => {
+          "external_related_links" => []
+        },
+        "links" => {
+        }
+      )
+
+      expected = {
+        related_items: [],
+        collections: [],
+        topics: [],
+        policies: [],
+        publishers: [],
+        other: [[], []]
+      }
+
+      expect(nothing).to eql(expected)
+    end
+
+    it "extracts and returns the appropriate related links" do
+      payload = payload_for("speech",
+        "details" => {
+          "body" => "body",
+          "government" => {
+            "title" => "government",
+            "slug" => "government",
+            "current" => true
+            },
+          "political" => true,
+          "delivered_on" => "2017-09-22T14:30:00+01:00"
+        },
+        "links" => {
+          "ordered_related_items" => [
+            {
+              "content_id" => "32c1b93d-2553-47c9-bc3c-fc5b513ecc32",
```

Comment:
I would limit the expected payload to the keys/values you actually assert on, or is it the case that you need the `content_id` and `locale` for validation purposes in `generator.merge_and_validate(content_item)`?

Reply:
Yep, it errored if these were not included.